### ISSUE TITLE
Feature Request: Enhancements to "Usage Browser"

### ISSUE THREAD

**Issue by hfloyd** [DanDiplo/Umbraco.GodMode#31]
Hi! It would be great if the "Usage Browser" could include templates as well, to see which Templates are actively being used on Content Nodes.

Perhaps with an additional "Schema Type" column for "Content Type" or "Template". Additionally, including a "Name" column (with searching) in addition to the "Alias" and "Description" columns.

Also, it seems that clicking on the item (which has a tooltip of "View related content") should pull up a filtered list of ONLY the content using that type. In my experience all the Content is displayed in the infinite editor fly-out.

(I'm using version 9.2.2)

**Comment by DanDiplo** [comment]
Hi Heather! Sounds like a good idea; I'll definitely look into this. Can't promise, as it depends on whether I can write the SQL!

**Comment by DanDiplo** [comment]
I've managed to fix the bug where the "view related content" wasn't filtering by the selected document type - so thanks for spotting that bug!

Unfortunately, I just can't seem to find way to incorporate templates into the browser. The Umbraco DB is complex and I couldn't figure out the joins without getting multiple rows returned. For instance, if a doc type had 2 templates it would always bring back both, not just the published one. I just can't figure out how it all joins together!

**Comment by hfloyd** [comment]
The way I've done is is by using the Umbraco APIs - the ContentService (IContent) does include info about the Template, I think. If you are doing all via direct SQL, though, I'm not sure how it connects...

**Comment by DanDiplo** [comment]
Yeah, the problem is the `ContentService` is too slow for what I'm doing; I basically have to use SQL group by to get a count of how many content items are used by a particular content type, and doing that via the `ContentService` would be extremely slow and require lots of calls. I'll keep digging!

**Comment by markadrake** [comment]
> Yeah, the problem is the `ContentService` is too slow for what I'm doing; I basically have to use SQL group by to get a count of how many content items are used by a particular content type, and doing that via the `ContentService` would be extremely slow and require lots of calls. I'll keep digging!

Looking at an Umbraco 10 database, I believe the `umbracoDocumentVersion` table has a `templateId` column. This column also has the `published` value (1 or 0). Hope I understood the above request correctly and this provides any insight at all.
